sda/internal/database/db_functions.go

```diff
@@ -622,3 +622,61 @@ func (dbs *SDAdb) getArchivePath(stableID string) (string, error) {
 
 	return archivePath, nil
 }
+
+// GetUserFiles retrieves all the files a user submitted
+func (dbs *SDAdb) GetUserFiles(userID string) ([]*SubmissionFileInfo, error) {
+	var err error
+
+	files := []*SubmissionFileInfo{}
+
+	// 2, 4, 8, 16, 32 seconds between each retry event.
+	for count := 1; count <= RetryTimes; count++ {
+		files, err = dbs.getUserFiles(userID)
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Duration(math.Pow(2, float64(count))) * time.Second)
+	}
+
+	return files, err
+}
+
+// getUserFiles is the actual function performing work for GetUserFiles
+func (dbs *SDAdb) getUserFiles(userID string) ([]*SubmissionFileInfo, error) {
```

Comment:
Is it over-complicating things to think about filtering or paging?

I suspect the primary users of the API for which this is provided are the ones who would upload hundreds of thousands or millions of files and could benefit from not having to grab all and filter locally.

Reply:
Down the road we at least need to limit this to files that doesn't have stable IDs yet.

As for filtering we could support POST request to the files endpoint with a path filter in the payload, but that is outside the scope of this PR.

```json
{"filter": "/subpath/target"}
```